main.go

```diff
@@ -98,10 +99,13 @@ func main() {
 	runMapper := func() error { return nil }
 	stopMapper := func(context.Context) error { return nil }
 	if flagData != "" && flagTrie != "" {
-		chain, err := chain.FromProtocolState(flagData)
+		opts := badger.DefaultOptions(flagData).WithLogger(nil)
```

Comment:
It's good to initialize all Badger database outside of the components, but we still need to use a customized set of options in order to limit the Badger memory usage. In terms of sequence, it would be good to first open all Badger databases before initializing our components.

Reply:
Solved by adding a `chain.DefaultOptions(dir string)` which generates the default options that we need in the chain package. WDYT?